packages/map-toolkit/src/deckgl/base-map/types.ts

```diff
@@ -124,4 +124,54 @@ export type MapViewportEvent = Payload<
   MapViewportPayload
 >;
 
-export type MapEventType = MapClickEvent | MapHoverEvent | MapViewportEvent;
+/**
+ * Payload for map control events that target a specific map instance.
+ * Used for enablePan, disablePan, enableZoom, disableZoom events.
+ */
+export type MapControlPayload = {
+  /** The map instance to apply the control change to */
+  id: UniqueId;
+};
+
+/**
+ * Payload for centerOn events to pan/zoom the map to a specific location.
+ */
+export type MapCenterOnPayload = {
+  /** The map instance to center */
+  id: UniqueId;
+  /** Longitude to center on */
+  longitude: number;
+  /** Latitude to center on */
+  latitude: number;
+  /** Optional zoom level */
+  zoom?: number;
+};
+
+export type MapEnablePanEvent = Payload<
+  typeof MapEvents.enablePan,
+  MapControlPayload
+>;
+
+export type MapDisablePanEvent = Payload<
+  typeof MapEvents.disablePan,
+  MapControlPayload
+>;
+
+export type MapEnableZoomEvent = Payload<
+  typeof MapEvents.enableZoom,
+  MapControlPayload
+>;
```

Comment:
take it or leave it: option to combine these into a single event like `MapEvents.togglePan` or MapEvents.setPan` and have an enabled: boolean flag.